packages/ts/react-crud/src/autoform-field.tsx

```diff
@@ -62,127 +94,117 @@ export type FieldOptions = Readonly<{
   validators?: Validator[];
 }>;
 
+type CommonFieldProps = Pick<
+  FieldOptions,
+  'className' | 'colspan' | 'disabled' | 'helperText' | 'id' | 'label' | 'placeholder' | 'readonly' | 'style'
+>;
+
+type FieldRendererProps = Readonly<{
+  model: AbstractModel;
+  field: FieldDirectiveResult;
+  fieldProps: CommonFieldProps;
+}>;
+
 function getPropertyModel(form: UseFormResult<any>, propertyInfo: PropertyInfo) {
   const pathParts = propertyInfo.name.split('.');
   // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
   return pathParts.reduce<any>((model, property) => (model ? model[property] : undefined), form.model);
 }
 
-type AutoFormTextFieldProps = SharedFieldProps & TextFieldProps;
-
-function AutoFormTextField({ propertyInfo, form, options, label, ...other }: AutoFormTextFieldProps) {
-  const model = getPropertyModel(form, propertyInfo);
-  return <TextField {...other} {...form.field(model)} label={label} />;
+function AutoFormTextField({ field, fieldProps }: FieldRendererProps) {
+  return <TextField {...field} {...fieldProps} />;
```

Comment:
Felt a bit wrong to just assign the `AutoFormFieldProps` to the Vaadin component, which includes properties like `options`, which the Vaadin component doesn't support. Likewise any other prop added to `AutoFormFieldProps` would end up being assigned to the component as well, which could lead to a bug at some point.

So I refactored the fields a bit so that they receive individual props for the model of the property, the `field` directive result and then the common `fieldProps` which come from `fieldOptions`.